homeassistant/components/plex/sensor.py

```diff
@@ -99,28 +99,16 @@ def __init__(
         verify_ssl,
     ):
         """Initialize the sensor."""
-        from plexapi.myplex import MyPlexAccount
-        from plexapi.server import PlexServer
-        from requests import Session
+        if plex_user or plex_password:
+            _LOGGER.error("Username and password no longer supported")
+            raise PlexException
 
         self._name = name
         self._state = 0
         self._now_playing = []
-
-        cert_session = None
-        if not verify_ssl:
-            _LOGGER.info("Ignoring SSL verification")
-            cert_session = Session()
-            cert_session.verify = False
-
-        if plex_token:
-            self._server = PlexServer(plex_url, plex_token, cert_session)
-        elif plex_user and plex_password:
-            user = MyPlexAccount(plex_user, plex_password)
-            server = plex_server if plex_server else user.resources()[0].name
-            self._server = user.resource(server).connect()
-        else:
-            self._server = PlexServer(plex_url, None, cert_session)
+        self._server = PlexServer(
+            {CONF_URL: plex_url, CONF_TOKEN: plex_token, CONF_VERIFY_SSL: verify_ssl}
+        )
 
```

Comment:
Don't we need to catch plex exceptions here? We do that in the media_player platform.

Reply:
Yes, that was already pending for another PR. Perhaps it makes sense to add it in here.

Reply:
Yeah I think we should add it here.

Reply:
It's caught a level up, but I've also added a catch for when the `sensor` calls update() similar to how it's done in `media_player`.

Reply:
We should have side effects in init method. Please move the creation of the server outise the entity init method. Pass the connected server instance to the entity instead.